src/Client/Web/Bit.Client.Web.BlazorUI/Components/BitComponentBase.cs

```diff
@@ -37,13 +35,29 @@ public override Task SetParametersAsync(ParameterView parameters)
                         Theme = (Theme)parameter.Value;
                         break;
 
+                    case nameof(IsEnabled):
+                        IsEnabled = (bool)parameter.Value;
+                        break;
+
                     case nameof(Style):
                         Style = (string)parameter.Value;
                         break;
 
                     case nameof(Class):
                         Class = (string)parameter.Value;
                         break;
+
+                    case nameof(Visibility):
+                        Visibility = (ComponentVisibility)parameter.Value;
+                        if (Visibility == ComponentVisibility.Hidden)
+                        {
+                            Style += ";visibility:hidden";
```

Comment:
What happens if I change Visibility multiple times?

Reply:
hmmm, your right.
lemme check.
it does not affect the desired functionality but it might append that phrase multiple times.

Reply:
I checked it and unfortunately it does append that phrase multiple times.
I'm working on a fix now.

Reply:
Maybe we need class mapper / style mapper which returns desired Style/Class as string based on added / removed classes styles.

Reply:
You're right, and I suggest  to discussed it in another issue.
I pushed a fix for this issue for now.